### ISSUE TITLE
Mainboard and toolhead board pinouts

### ISSUE THREAD

**Issue by 53Aries** [QIDITECH/QIDI_Q1_Pro#30]
Any chance we can have the pinout diagram and/or schematic for both the main board and toolhead board? Thanks

**Comment by CChen616** [comment]
Here you go
[A-8(RP) V1.0_001 PIN.pdf](https://github.com/QIDITECH/QIDI_Q1_Pro/files/15463427/A-8.RP.V1.0_001.PIN.pdf)
[X-7 V1.1_001 PIN.pdf](https://github.com/QIDITECH/QIDI_Q1_Pro/files/15463428/X-7.V1.1_001.PIN.pdf)

**Comment by 53Aries** [comment]
My unit has A-7 toolhead board. Do you have a pinout for that?

> Here you go [A-8(RP) V1.0_001 PIN.pdf](https://github.com/QIDITECH/QIDI_Q1_Pro/files/15463427/A-8.RP.V1.0_001.PIN.pdf) [X-7 V1.1_001 PIN.pdf](https://github.com/QIDITECH/QIDI_Q1_Pro/files/15463428/X-7.V1.1_001.PIN.pdf)

**Comment by CChen616** [comment]
[A-7 V1.0_002 PIN.pdf](https://github.com/user-attachments/files/15519717/A-7.V1.0_002.PIN.pdf)

**Comment by 53Aries** [comment]
Can you tell me if on the Q1 the A-7 toolhead board is using CAN or USB?

**Comment by CChen616** [comment]
It's using serial, and here is the make menuconfig for the toolhead, it might require using STM32-Link to flash

![image](https://github.com/QIDITECH/QIDI_Q1_Pro/assets/88018764/f9009e60-f986-45a0-a92a-b071523248eb)

**Comment by 53Aries** [comment]
> It's using serial, and here is the make menuconfig for the toolhead, it might require using STM32-Link to flash
> 
> ![image](https://private-user-images.githubusercontent.com/88018764/335780837-f9009e60-f986-45a0-a92a-b071523248eb.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTcyNDc2OTQsIm5iZiI6MTcxNzI0NzM5NCwicGF0aCI6Ii84ODAxODc2NC8zMzU3ODA4MzctZjkwMDllNjAtZjk4Ni00NWEwLWE5MmEtYjA3MTUyMzI0OGViLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA2MDElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNjAxVDEzMDk1NFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWVmYzA4ZjM0OTZkMjNiOTc1YWYyMGUxNzM1YzAwYzUwZTY3NDkxYTdkMDA3ZThjMGQ1ZGQ4MTBkYjkzZWE1NWUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.YKTF4CcHbka1QiCs4HltTBlayM3w08F0JBgJ3cCm5Yc)

I believe this is for the mainboard. The toolhead is an rp2040.

**Comment by CChen616** [comment]
Ah, yes, I confused it with another model

**Comment by 53Aries** [comment]
> Yes, I confused it with another model

No worries. I ask because I'm unable to find the rp2040 toolhead board listed in the linux environment. It doesn't show as a usb or serial device. Wondering how the x7 mainboard host is configured to connect with the A7 toolhead. In the qidi config the toolhead mcu is listed as ttyS2 so i assume serial.

**Comment by Wuehli** [comment]
Hi, 
is there also a pinout for the X-7 v1.0, as it seems to be different from the 1.1 you shared above.

Additional question, is there a possibility to bypass the SOC and connect e.g. a Raspberry to the STM32F402 MCU?
Thanks!!

**Comment by CChen616** [comment]
[X-7 V1.0_001 PIN.pdf](https://github.com/user-attachments/files/15746098/X-7.V1.0_001.PIN.pdf)

> Additional question, is there a possibility to bypass the SOC and connect e.g. a Raspberry to the STM32F402 MCU?
> Thanks!!

Not sure 😢

**Comment by Seviged** [comment]
> Hi, is there also a pinout for the X-7 v1.0, as it seems to be different from the 1.1 you shared above.
> 
> Additional question, is there a possibility to bypass the SOC and connect e.g. a Raspberry to the STM32F402 MCU? Thanks!!

Technically, it is possible. The STM32F401 uses PA11 (D-) and PA12 (D+) for communication through the USB interface. The X7 v1.0 board has a m-USB connector wired to PA11 and PA12, but the connector is not soldered, and it may lack the 22-30 ohm resistors (I don't have this motherboard yet, and the X7 doesn’t have any schematics to confirm my assumptions).

If it is possible to solder the missing components on this motherboard, you can flash Klipper (or better yet, [Katapult](https://github.com/Arksine/katapult)) to the STM32 using an SD card. In the menuconfig, choose USB (PA11, PA12). After flashing Katapult via the SD card, you can flash Klipper without the SD card directly from the SSH console.

The X7 v1.1 doesn't have a micro-USB connector onboard, but the PA11 and PA12 pins are still present at the FAN4 and FAN5 connectors. In this case, you might need to desolder some components and build the USB connection "from scratch."

All the above steps might be necessary if you want to connect via USB, essentially to a main PC or home server with an AMD64 architecture.

Another option is to use the PA09 and PA10 pins (USART1). According to the pinout, these lines have 100-ohm resistors. You could desolder them, attach some wires to the MCU (STM32) side, and connect it to another UART 3.3V-compatible interface through new 100-ohm resistors. This option is mainly for ARM64 devices (like the Raspberry Pi). You can also connect it to a USB-UART converter (preferably one with an original FTDI chip).

All of the above is theoretical; I haven't tried it yet. I’m a bit disappointed that Qidi didn’t mention the possibilities with this motherboard and seems to have cut off the MCU USB interface in the 1.1 version.

**Comment by 53Aries** [comment]
> > Hi, is there also a pinout for the X-7 v1.0, as it seems to be different from the 1.1 you shared above.
> > Additional question, is there a possibility to bypass the SOC and connect e.g. a Raspberry to the STM32F402 MCU? Thanks!!
> 
> Technically, it is possible. The STM32F402 uses PA11 (D-) and PA12 (D+) for communication through the USB interface. The X7 v1.0 board has a m-USB connector wired to PA11 and PA12, but the connector is not soldered, and it may lack the 22-30 ohm resistors (I don't have this motherboard yet, and the X7 doesn’t have any schematics to confirm my assumptions).
> 
> If it is possible to solder the missing components on this motherboard, you can flash Klipper (or better yet, [Katapult](https://github.com/Arksine/katapult)) to the STM32 using an SD card. In the menuconfig, choose USB (PA11, PA12). After flashing Katapult via the SD card, you can flash Klipper without the SD card directly from the SSH console.
> 
> The X7 v1.1 doesn't have a micro-USB connector onboard, but the PA11 and PA12 pins are still present at the FAN4 and FAN5 connectors. In this case, you might need to desolder some components and build the USB connection "from scratch."
> 
> All the above steps might be necessary if you want to connect via USB, essentially to a main PC or home server with an AMD64 architecture.
> 
> Another option is to use the PA09 and PA10 pins (USART1). According to the pinout, these lines have 100-ohm resistors. You could desolder them, attach some wires to the MCU (STM32) side, and connect it to another UART 3.3V-compatible interface through new 100-ohm resistors. This option is mainly for ARM64 devices (like the Raspberry Pi). You can also connect it to a USB-UART converter (preferably one with an original FTDI chip).
> 
> All of the above is theoretical; I haven't tried it yet. I’m a bit disappointed that Qidi didn’t mention the possibilities with this motherboard and seems to have cut off the MCU USB interface in the 1.1 version.

On this topic, I was poking around and found the pads on the rear of the board where PA11 and 12 would have the 22ohm resistors and confirmed they are connected at the MCU. Looking at the stm32 data sheet I don't know that having the resistors is necessary, but that's above my head right now. My question is what about ground and shield? The upper portion of the smaller pads on the left and right of PA11/12 are connected to ground. The larger center pads are PA11/12. @CChen616 could we pleeeeeease get the schematic for x-7 1.0? It's old enough at this point that surely @QIDITECH wont sustain any loss from sharing it. 
![20250104_003447](https://github.com/user-attachments/assets/16c457f6-646c-47fd-a9fb-5e204a44a445)
Since we don't need to provide power on the USB port, could I not just connect he D+ and D- to the PA11/12 pads ignoring the resistors? I'm sure ground and shield should be connected to something though. 
Basically would this work, minus the power. 
![Screenshot 2025-01-04 024422](https://github.com/user-attachments/assets/cc148e35-a15b-4194-8a19-bd4855c65191)

**Comment by 53Aries** [comment]
I have confirmed you can connect to the PA11/12 pads on the rear of the board with D+ and D- of a usb cable, ground, flash the stm with klipper built for USB communication and it works. The inbuilt USB of the STM32F4 does not need 22ohm resistors.
![20250104_145633](https://github.com/user-attachments/assets/5e12193f-76b1-4884-b932-963b9ad9ccc7)
Rename the klipper.bin to qd_mcu.bin
![Screenshot 2025-01-04 163109](https://github.com/user-attachments/assets/a7d2b820-f22b-4b4e-a3cd-f2547367063c)
![Screenshot 2025-01-04 163722](https://github.com/user-attachments/assets/74ff7705-6541-4931-9646-e2059b533ebd)
![Screenshot 2025-01-04 163728](https://github.com/user-attachments/assets/73de4f52-63dd-4fa0-83d5-1e9be6da6d6c)

**Comment by Bytedecode72** [comment]
does anyone know if there are full circuit board schematics for the adapter plates A7-v1.0 (q1 pro) that when the front cover fell off it snapped the cables to the cooling fan and they shorted out, the fan survived but there is now no power on the green socket on the adapter plate so have ordered another one but would like to fix the original and odds are it will be one maybe two componets that died as a result if not i see tracing and alot of testing to do

**Comment by dewi-ny-je** [comment]
> Yes, I confused it with another model

so which USART pins are the correct ones for the Q1 toolhead? UART0 on GPIO0/1?

Thanks
